src/webgpu/api/validation/capability_checks/limits/maxStorageBufferBindingSize.spec.ts

```diff
@@ -0,0 +1,166 @@
+  testValueName: TestValue,
+  defaultLimit: number,
+  maximumLimit: number
+) {
+  const requestedLimit = getDeviceLimitToRequest(limitValueTest, defaultLimit, maximumLimit);
+  return {
+    requestedLimit,
+    testValue: getTestValue(testValueName, requestedLimit),
+  };
+}
+
+const limit = 'maxStorageBufferBindingSize';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBindGroup,at_over')
+  .desc(`Test using createBindGroup at and over ${limit} limit`)
+  .params(kLimitBaseParams.combine('bufferPart', kBufferPartsKeys))
+  .fn(async t => {
+    const { limitTest, testValueName, bufferPart } = t.params;
+    const { adapter, defaultLimit, maximumLimit } = await t.getAdapterAndLimits();
+    const { requestedLimit, testValue } = getDeviceLimitToRequestAndValueToTest(
+      limitTest,
+      testValueName,
+      defaultLimit,
+      maximumLimit
+    );
+
+    await t.testDeviceWithSpecificLimits(
+      adapter,
+      requestedLimit,
+      testValue,
+      async ({ device, testValue, shouldError }) => {
+        const bindGroupLayout = device.createBindGroupLayout({
+          entries: [
+            {
+              binding: 0,
+              visibility: GPUShaderStage.COMPUTE,
+              buffer: { type: 'storage' },
+            },
+          ],
+        });
+
+        const { size, offset } = getSizeAndOffsetForBufferPart(device, bufferPart, testValue);
+        device.pushErrorScope('out-of-memory');
+        const storageBuffer = device.createBuffer({
+          usage: GPUBufferUsage.STORAGE,
+          size,
+        });
+        const outOfMemoryError = await device.popErrorScope();
+
+        if (!outOfMemoryError) {
+          await t.expectValidationError(
+            () => {
+              device.createBindGroup({
+                layout: bindGroupLayout,
+                entries: [
+                  {
+                    binding: 0,
+                    resource: {
+                      buffer: storageBuffer,
+                      ...(offset && { offset }),
+                      size: testValue,
+                    },
+                  },
+                ],
+              });
+            },
+            shouldError,
+            `size: ${size}, offset: ${offset}, testValue: ${testValue}`
+          );
+
+          storageBuffer.destroy();
+        }
+      }
+    );
+  });
+
+g.test('validate,maxBufferSize')
+  .desc(`Test that ${limit} <= maxBufferSize`)
+  .params(u => u.combine('limitTest', kLimitValueTestKeys))
+  .fn(async t => {
+    const { limitTest } = t.params;
+    const { adapter, defaultLimit, maximumLimit } = await t.getAdapterAndLimits();
+    const requestedLimit = getDeviceLimitToRequest(limitTest, defaultLimit, maximumLimit);
+
+    await t.testDeviceWithSpecificLimits(adapter, requestedLimit, 0, ({ device, actualLimit }) => {
+      t.expect(actualLimit <= device.limits.maxBufferSize);
+    });
+  });
+
+g.test('validate,greaterThanOrEqualTo32')
```

Comment:
I can't seem to find this restriction explicitly in the spec? Is this an implication of the min*BufferOffsetAlignment requirements? (If it is then it's fine, just wanted to understand, and maybe would be useful to add in the description so it's more obvious). Ditto for the other file.

Reply:
the limit is here: https://www.w3.org/TR/webgpu/#dom-supported-limits-minuniformbufferoffsetalignment

Reply:
strike that, I think you're right. I have no idea where I got that limit from 😅

Reply:
deleted test